doc/nrf/gs_modifying.rst

```diff
@@ -150,9 +220,32 @@ See :ref:`zephyr:menuconfig` in the Zephyr documentation for instructions on how
 To locate a specific configuration option, use the filter (:guilabel:`Jump to` in menuconfig and guiconfig).
 The documentation for each :ref:`configuration option <configuration_options>` also lists the menu path where the option can be found.
 
-.. important::
-   All changes to the :file:`.config` file are lost when you clean your build directory.
-   You can save it to another location, but you must then manually copy it back to your build directory.
+Changing the configuration permanently
+======================================
+
+To configure your application and maintain the configuration when you clean the build directory, you need to specify the configuration in one of the permanent configuration files.
+In most cases, this will be either the default :file:`prj.conf` file of the application or an extra Kconfig fragment.
+In these files, you can specify different values for configuration options that are defined by a library or board, and you can add configuration options that are specific to your application.
+
+See :ref:`zephyr:setting_configuration_values` in the Zephyr documentation for information on how to change the configuration permanently.
+
+.. tip::
+   Modifying the temporary configuration can result in an invalid or bloated configuration, because the default values have already been calculated based on the previous configuration.
```

Comment:
This give the impression that we accept that using `menuconfig` results in an invalid configuration.

I believe that if user is using `menuconfig` to try out temporary changes, then that should not result in `invalid` configuration just because `menuconfig` was used.

If it does, then it's a (dependency) bug and we should fix it.

That said, disabling bluetooth stack on a sample which requires bluetooth will of course render the sample invalid, but that has nothing to do with using `menuconfig` cause the same will happen if you disable the bluetooth stack directory in `prj.conf`.

```suggestion
   Modifying the temporary configuration can result in a bloated configuration, because disabling a setting may not disable other settings that are no longer required, see https://docs.zephyrproject.org/latest/guides/build/kconfig/tips.html#stuck-symbols-in-menuconfig-and-guiconfig
```
Link: https://docs.zephyrproject.org/latest/guides/build/kconfig/tips.html#stuck-symbols-in-menuconfig-and-guiconfig

Reply:
This is how this tip would look like:

>    Modifying the temporary configuration can result in a bloated configuration, because disabling a setting may not disable other settings that are no longer required.
>    See the "Stuck symbols in menuconfig and guiconfig" section on the :ref:`zephyr:kconfig_tips_and_tricks` page in the Zephyr documentation for more information.
>    It is a good practice to modify :file:`prj.conf` instead of modifying the configuration in menuconfig.

I'll wait for the decision of the discussion here: https://github.com/nrfconnect/sdk-nrf/pull/6121#discussion_r763832584